### PR TITLE
Gives AIs a command to unlock their core cover.

### PR DESCRIPTION
**Pull request by Paaiii**
## About the PR 
Adds a verb to AI commands that allows the AI player to unlock or relock their own core cover, similar to cyborgs.

## Why's this needed? 
Cyborgs already have a command to do this with their cover, and the fact that the AI can't unlock themselves can be a consistent pain, especially if there aren't any command.

## Changelog 

```changelog
(u)Paaiii
(+)Gives AIs a command to unlock their own core cover.
```

### CODE REVIEW

File: code/mob/living/silicon/ai.dm [hunk 1]
```diff
@@ -126,6 +126,7 @@ var/global/list/ai_emotions = list("Annoyed" = "ai_annoyed-dol", \
 	var/moustache_mode = 0
 	var/status_message = null
 	var/mob/living/silicon/deployed_shell = null
+	var/locking = 0
```
Comment (Tarmunora) [inline code/mob/living/silicon/ai.dm:129]: If you're going to mirror this var onto AI frames you should also mirror how it's used - i.e.: resetting to `0` when manually (un)locking, or popping open with tools/etc.

Comment (Paaiii) [reply code/mob/living/silicon/ai.dm:129]: Good catch- should be sorted now!

File: code/mob/living/intangible/ai-camera.dm [hunk 2]
```diff
@@ -453,6 +453,29 @@
 		if(mainframe)
 			mainframe.deploy_to()
 
+	verb/toggle_lock()
+		set category = "AI Commands"
+		set name = "Toggle Cover Lock"
+		if(mainframe)
```
Comment (TobleroneSwordfish) [inline code/mob/living/intangible/ai-camera.dm:459]: Why not just call the proc on the mainframe?

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/mob/living/intangible/ai-camera.dm (hunk 2 above)
Comment (Paaiii) [reply code/mob/living/intangible/ai-camera.dm:459]: .....very good question actually

File: code/mob/living/silicon/ai.dm [hunk 3]
```diff
@@ -505,6 +508,8 @@ or don't if it uses a custom topopen overlay
 					src.dismantle_stage = 0
 				else
 					src.dismantle_stage = 1
+				if (src.locking)
+					src.locking = 0
```
Comment (frawhst) [inline code/mob/living/silicon/ai.dm:512]: You don't need this check, `locking` is going to be `0` after either way
```suggestion
				src.locking = 0
```

File: code/mob/living/silicon/ai.dm [hunk 4]
```diff
@@ -448,6 +449,8 @@ or don't if it uses a custom topopen overlay
 		if (src.dismantle_stage == 1)
 			playsound(src.loc, 'sound/items/Crowbar.ogg', 50, 1)
 			src.visible_message("<span class='alert'><b>[user.name]</b> opens [src.name]'s chassis cover.</span>")
+			if (src.locking)
+				src.locking = 0
```
Comment (frawhst) [inline code/mob/living/silicon/ai.dm:453]: Same thing
```suggestion
			src.locking = 0
```

**Review by frawhst — COMMENTED** [review]
Yeah I got annoyed that I couldn't do this recently.
Couple small things.

**Pull request merged**